posts/views.py

```diff
@@ -241,27 +242,38 @@ def post(self, request):
         # handle form data for categories
         if type(request.data) is dict:
             categories = request.data.get("categories")
+            visibleTo = request.data.get("visibleTo")
         else:
             categories = request.data.getlist("categories")
+            visibleTo = request.data.getlist("visibleTo")
 
-        if categories is not None:
-            # author has defined categories
-            for cat in categories:
-                # check if category exists
-                try:
-                    cat_obj = Category.objects.get(category=cat)
-                except Category.DoesNotExist:
-                    cat_obj = None
-
-                if cat_obj is None:
-                    Category.objects.create(category=cat)
+        # TODO: check if atomic is frowned upon
```

Comment:
From my understanding I think this is blocking and creating any categories/viewers before creating the post. I don't see an issue since this prevents nested serialization problems that we would otherwise have. Please cite any sources that you used for this.